parser_library/src/checking/instr_operand.cpp

```diff
@@ -287,6 +287,13 @@ bool one_operand::check(
         }
         return true;
     }
+    // simple operand with relocatable symbol or immediate value
+
+    if (to_check.identifier.type == machine_operand_type::RELOC_IMM && operand_identifier != "RELOC")
```

Comment:
This is basically reusing the `operand_identifier` as an flag, right? I suggest introducing that flag instead.